src/sage/plot/misc.py

```diff
@@ -139,6 +139,13 @@ def setup_for_eval_on_grid(funcs,
     else:
         vars, free_vars = unify_arguments(funcs)
 
+    #check for invalid range entered by user
+    if (ranges[0][-2]>ranges[0][-1]):
+        raise ValueError("xrange not correctly defined: xmin(={}) > xmax(={})".format(ranges[0][-2], ranges[0][-1]))
+
+    if (ranges[1][-2]>ranges[1][-1]):
+        raise ValueError("xrange not correctly defined: ymin(={}) > ymax(={})".format(ranges[1][-2], ranges[1][-1]))
+
```

Comment:
needs spaces around `>`. It seems you can use `[0]`, `[1]` instead of `[-2]` and [-1]` respectively since ranges is normalized by now.

Reply:
Spaces added.

Have to use the minus method as range can be entered as (xmin, xmax) or (x,xmin,xmax) last I checked.